server/bin/pbench-dispatch.sh

```diff
@@ -191,11 +195,29 @@ while read tbmd5; do
     popd > /dev/null 2>&4
     if [[ ${sts} -ne 0 ]]; then
         log_error "${TS}: Quarantined: ${tb} failed MD5 check" "${status}"
-        quarantine ${quarantine}/${controller} ${tb} ${tb}.md5
+        quarantine ${quarantine}/${controller} ${tbmd5} ${tb}
         (( nquarantined++ ))
         continue
     fi
 
+    if [[ -n "${backup_dir}" ]]; then
+        # Backup to the local backup directory.
+        mkdir -p ${backup_dir}/${controller}
+        sts=${?}
+        if [[ ${sts} -ne 0 ]]; then
+            log_error "${TS}: failed to create backup directory ${backup_dir}/${controller}: code ${sts}" "${status}"
+            (( nberrs++ ))
+            continue
+        fi
+        cp -a ${tbmd5} ${tb} ${backup_dir}/${controller}/
+        sts=${?}
+        if [[ ${sts} -ne 0 ]]; then
+            log_error "${TS}: failed to backup ${tbmd5} and ${tb}: code ${sts}" "${status}"
+            (( nberrs++ ))
+            continue
+        fi
+    fi
+
```

Comment:
Do we really not want to do anything with the tar ball if we cannot back it up? I would think we want it to go through normal processing to allow users to look at their data, even if we have to fix the backup behind the scenes.

Reply:
It is a major issue to not have a tar ball backed up.  So we need to address it first.

By leaving it in the reception area which make it easy to ensure the system is self-consistent: fixing the backup configuration allows data to flow correctly with it being backed up.

If we allow dispatch to move data into place without backing it up, since we have removed the backup cron job we have no way to "re-backup" the data without manual intervention.

Reply:
That makes sense: I guess that's the price we have to pay for the simplification.